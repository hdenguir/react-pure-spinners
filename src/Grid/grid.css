:root {
  --rps-grid-color: #fff;
  --rps-grid-size: 64px;
}

.rps-grid {
  display: inline-block;
  position: relative;
  width: var(--rps-grid-size);
  height: var(--rps-grid-size);
}

.rps-grid div {
  position: absolute;
  width: calc(var(--rps-grid-size) / 5);
  height: calc(var(--rps-grid-size) / 5);
  border-radius: 50%;
  background: var(--rps-grid-color);
  animation: rps-grid 1.2s linear infinite;
}

.rps-grid div:nth-child(1) {
  top: calc(var(--rps-grid-size) / 10);
  left: calc(var(--rps-grid-size) / 10);
  animation-delay: 0s;
}

.rps-grid div:nth-child(2) {
  top: calc(var(--rps-grid-size) / 10);
  left: calc(var(--rps-grid-size) / 2.5);
  animation-delay: -0.4s;
}

.rps-grid div:nth-child(3) {
  top: calc(var(--rps-grid-size) / 10);
  left: calc(var(--rps-grid-size) / 1.42857);
  animation-delay: -0.8s;
}

.rps-grid div:nth-child(4) {
  top: calc(var(--rps-grid-size) / 2.5);
  left: calc(var(--rps-grid-size) / 10);
  animation-delay: -0.4s;
}

.rps-grid div:nth-child(5) {
  top: calc(var(--rps-grid-size) / 2.5);
  left: calc(var(--rps-grid-size) / 2.5);
  animation-delay: -0.8s;
}

.rps-grid div:nth-child(6) {
  top: calc(var(--rps-grid-size) / 2.5);
  left: calc(var(--rps-grid-size) / 1.42857);
  animation-delay: -1.2s;
}

.rps-grid div:nth-child(7) {
  top: calc(var(--rps-grid-size) / 1.42857);
  left: calc(var(--rps-grid-size) / 10);
  animation-delay: -0.8s;
}

.rps-grid div:nth-child(8) {
  top: calc(var(--rps-grid-size) / 1.42857);
  left: calc(var(--rps-grid-size) / 2.5);
  animation-delay: -1.2s;
}

.rps-grid div:nth-child(9) {
  top: calc(var(--rps-grid-size) / 1.42857);
  left: calc(var(--rps-grid-size) / 1.42857);
  animation-delay: -1.6s;
}

@keyframes rps-grid {
  0%,
  100% {
    opacity: 1;
  }
  50% {
    opacity: 0.5;
  }
}
