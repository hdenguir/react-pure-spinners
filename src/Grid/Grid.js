import './grid.css'
import React from 'react'
import PropTypes from 'prop-types'
import { clsx } from '../utils'

const Grid = ({ className, color, size, style }) => {
  const varStyle = { ...style }
  color && (varStyle['--rps-grid-color'] = color)
  size && (varStyle['--rps-grid-size'] = `${size}px`)

  return (
    <div className={clsx('rps-grid', className)} style={varStyle}>
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
    </div>
  )
}

Grid.propTypes = {
  className: PropTypes.string,
  color: PropTypes.string,
  size: PropTypes.number,
  style: PropTypes.object
}

export default Grid
