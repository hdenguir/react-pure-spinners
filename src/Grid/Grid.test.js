import React from 'react'
import renderer from 'react-test-renderer'
import Grid from './Grid'

it('renders correctly', () => {
  const tree = renderer.create(<Grid />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('joins default class with className prop', () => {
  const { root } = renderer.create(<Grid className="spin" />)

  expect(root.findByType('div').props.className).toEqual('rps-grid spin')
})

it('sets CSS vars with props', () => {
  const { root } = renderer.create(
    <Grid size={100} color="#d36ac2" style={{ margin: 20 }} className="spin" />
  )
  const { style } = root.findByType('div').props

  expect(style).toEqual({
    '--rps-grid-size': '100px',
    '--rps-grid-color': '#d36ac2',
    margin: 20
  })
})
