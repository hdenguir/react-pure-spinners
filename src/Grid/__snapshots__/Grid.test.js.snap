// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="rps-grid"
  style={Object {}}
>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;
